src/shogun/classifier/AveragedPerceptron.cpp

```diff
@@ -40,7 +40,7 @@ void AveragedPerceptron::init()
 	    ParameterProperties::MODEL)
 }
 
-void AveragedPerceptron::init_model(std::shared_ptr<Features> data)
+void AveragedPerceptron::init_model(const std::shared_ptr<Features>& data)
```

Comment:
unrelated to this effort...

Reply:
I had to do this as I had made a change in IterativeMachine as you had said: https://github.com/shogun-toolbox/shogun/pull/5006#discussion_r415634474

And then AveragedPerceptron, Perceptron, and NewtonSVM are derived from IterativeMachine and so had to make corresponding changes there to prevent compile-time errors.

Reply:
this seems to be a rebase issue ... the compiler errors should have been fixed by others